### ISSUE TITLE
Component List Panel View

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#4431]
This task is automatically imported from the old Task Issue Board and it was originally created by Ilya Bogdanov.
Original issue is [here](https://www.pivotaltracker.com/story/show/180892146).
<hr />

## Why

**As** Enso User
**I want** to see all available components in a predictable and easy-to-use layout grouped by their module/purpose
**So that** I can quickly find and access needed components using Component Browser

## Description

This task covers the view of Component List Panel: the component groups arranged in three sections inside the scroll area. The sections are, from bottom to top:
* Favorites Data Science Tools
* Local Scope
* Sub-modules
The Local Scope has a single, three-column component group, others have many single-column. The layout of the latter groups should be roughly from left-to-right, from bottom-to-top, keeping also all columns in similar size. The component groups and their order shall be defined by the Component List Panel user (the developer).

What is not in scope:
* Navigating with keyboard,
* Left-side panel,
* ordering of Virtual Component Groups.

## Acceptance Criteria

```gherkin
Scenario: 
GIVEN There is a demo scene featuring Component Panel View
WHEN I open this scene
THEN I see an example Component Panel composed of several Component Groups
AND The bottom-most item of the bottom-middle Component Group is an Active Focus Item

WHEN There is an Active Focus Item in one of the Component Groups
THEN There are no focused items in any other Component Group (there is exactly one Active Focus Item at all times)
```

## Tasks:
- [X] Basic shape layout
- [X] API + Visuals for Favorites Data Science Tools
- [X] API + Visuals for Sub-modules
- [X] API + Visuals for Local Scope
- [X] Enable correct scrolling behaviour
- [ ] Enable correct selection handling
- [X] Design Clean-up

## Comments:
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-05-17):

**Progress:** Started looking into state of different components and read up on the design document for the Component List Panel. Some testing of the the new build scripts. It should be finished by 2022-06-03.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Get started on implementing the basic layout of the Component List Panel and further investigate API requirements, which data to sue and what to mock. (Enso Bot - May 18, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-05-18):

**Progress:** Started on the implementation of Component List Panel. Created demo scene and structs for showing the basic panel. Researched current state of required components (i.e. Component Group) and the structure of their demo scenes. It should be finished by 2022-06-03.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Add first basic API for adding Component Groups to the Component List Panel. Discuss scope/sub-task splitting. (Enso Bot - May 19, 2022)
<hr />
**Technical Design** review set to **pass** (Adam Obuchowicz - May 19, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-05-19):

**Progress:** Started planning of sub-tasks and checking against existing impl of sub-components. Started working on the adding of API for adding sub-components and crumbs. It should be finished by 2022-06-03.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Finalise plan for sub-tasks. Focus on first task, probably crumbs  (top panel) implementation. (Enso Bot - May 20, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for the provided date (2022-05-20):

**Progress:** Finished the basic layout and investigation of the details of the required section behaviour. Coordinates with Adam and Ilya about working son some tasks in parallel (and further clarifying scope and already  existing features). It should be finished by 2022-06-03.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). I'll prep the first draft PR for the first sub-task with demo, coordinate with Ilya about progress, do planning prep for Performance Team. (Enso Bot - May 23, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-05-23):

**Progress:** Planning and prep for meetings.  Design and Code cleanup for first draft PR. It should be finished by 2022-06-03.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). More coordination with Ilya on collaboration on code. Finalise draft PR for pre-review and get started on implementing sections. (Enso Bot - May 24, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-05-24):

**Progress:** Finalising the current draft PR is taking longer than expected. While the shapes etc were easy enough, working with the clipping area turns out to be more complicated. Need tor revisit the shape and layer handling of the involved shapes to make this work. It should be finished by 2022-06-03.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Continue work with the clipping area and updating shapes and components as needed. (Enso Bot - May 25, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-05-25):

**Progress:** Still working on shapes and clipping. This is weird and I am working with Ilya to resolve it. The ergonomics of using our scroll area are less than ideal. It should be finished by 2022-06-03.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Talk again with Ilya and solve this issue. Then finally create the draft PR and start working on sections. (Enso Bot - May 26, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-05-26):

**Progress:** Resolved the issue with Ilya. Now updating and cleaning the code, also updating the descriptions of the clipping area and thinking about whether the API could be more ergonomic / less error prone. It should be finished by 2022-06-03.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Finish the draft PR with the working clipping area and start work on sections. (Enso Bot - May 27, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for today (2022-05-30):

**Progress:** Finalised first darft PR. Got started ona dding facilities for sections, which will lead to the proper layout of the "Favorite Data Science Tools" section. It should be finished by 2022-06-03.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Finish the layout of the "Favorite Data Science Tools" and update draft PR. Discuss API for the different sections with Adam. (Enso Bot - May 30, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for today (2022-05-31):

**Progress:** Integrated designs for sections (headings layout, margins backgrounds). Encountered an issue were text is changing layer when setting font, and working on fixing that. It should be finished by 2022-06-03.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Finish the section layout and update draft PR with that. (Enso Bot - May 31, 2022)
<hr />
**Michael Mauderer** reports a new **🔴 DELAY** for today (2022-06-02):

**Summary:** There is **5 days** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 3 days delay** for the delivery of this weekly plan.

2 days weekend.

**Delay Cause:** Work took longer than expected due to shape ordering / layering issues and missing functionality to the existing APIs. (Enso Bot - Jun 2, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-01):

**Progress:** Fixed issue about ordering and layering of UI elements within the component. Further integrated designs. Noticed an issue with the rounded corners of the scroll area (we need only the lower right corner to be rounded, but the area only allows all of the corners to be rounded.) and extended the scroll area to allow for this. It should be finished by 2022-06-08.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Check the design matches, then update draft PR. Continue with next sections. (Enso Bot - Jun 2, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-02):

**Progress:** Fixed some design mismatches. Encountered issue with the scroll mask and started investigating. It should be finished by 2022-06-08.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Update draft PR. Continue with next sections. (Enso Bot - Jun 3, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for the provided date (2022-06-03):

**Progress:** Implemented the basic API and visuals for all three sections. Needs some cleanup visually and code wise, then I can update the draft PR. This should be doable Monday before planning. It should be finished by 2022-06-08.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Clean up the sections code, and visuals. Start looking at integration with Ilya/Adam and/or scrolling behaviour. (Enso Bot - Jun 5, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-06):

**Progress:** Cleaned up the section code and made the visuals presentable. It should be finished by 2022-06-08.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Start working on scrolling, discuss header placement. (Enso Bot - Jun 7, 2022)
<hr />
**Michael Mauderer** reports a new **🔴 DELAY** for yesterday (2022-06-07):

**Summary:** There is **2 days** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Work took longer than expected. (Enso Bot - Jun 8, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-07):

**Progress:** Did some more cleanup of the draft PR for review. Started investigating scrolling behaviour changes and existing APIs. Got started on implementation. It should be finished by 2022-06-10.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Finish working on scrolling. Coordinate integration requirements / or start design cleanup. (Enso Bot - Jun 8, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-08):

**Progress:** Implemented review feedback for PR. Implemented headers staying in place while scrolling. It should be finished by 2022-06-10.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Align scrolling behaviour further with design doc (scroll bar, event capture) and implement culling/disabling of elements that are not visible. (Enso Bot - Jun 9, 2022)
<hr />
**Michael Mauderer** reports a new **🔴 DELAY** for yesterday (2022-06-09):

**Summary:** There is **3 days** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 0 days delay** for the delivery of this weekly plan.

2 days weekend.

**Delay Cause:** Additional review feedback and design work. Also some underlying functionality needs to be further extended: scroll area needs additional work to make it work as required by the design (.i.e., adjusting the scrollbar appearance and behaviour). (Enso Bot - Jun 10, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-09):

**Progress:** Finished work on scrolling behaviour: headers stay in place, Sub-components not visible get disconnected, scrolling events are handled correctly. Implemented initial review feedback on the intermediary PR. It should be finished by 2022-06-13.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Implement second round of PR feedback, start work on the outstanding design fixes (colours/alignment/style and scrollbar behaviour/look) (Enso Bot - Jun 10, 2022)
<hr />
**Adam Obuchowicz** reports a new **STANDUP** for the last Friday (2022-06-10):

**Progress:** Did some reviews in the morning, then switched from integration to work on Component View; added there the component-group::set::Wrapper and connect its outputs to the Component List Panel outputs. It should be finished by 2022-06-13.

**Next Day:** Next day I will be working on the [##181869402 task](https://www.pivotaltracker.com/story/show/181869402). Integrate the freshly created output integration + Documentation viewer. (Enso Bot - Jun 13, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for the provided date (2022-06-10):

**Progress:** Worked on second round of review fixes. This takes a bit more time, as lots of things need to be changed to the theme, which is a bit tedious. (Note we might want to create a macro for some of this,still thinking about how this could look). It should be finished by 2022-06-13.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Finish the PR fixes. Go back to fixing the design issues. (Enso Bot - Jun 13, 2022)
<hr />
**Michael Mauderer** reports a new **🔴 DELAY** for yesterday (2022-06-13):

**Summary:** There is **2 days** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Refactoring to use the theme and resulting cleanup took longer than expected. (Enso Bot - Jun 14, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-13):

**Progress:** Almost finished transition from constants to theme. Required some rejiggling of lots of functionality and debugging the resulting changes. Should just require a bit of cleanup to be ready for re-review. It should be finished by 2022-06-15.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Clean the PR for re-review. Write up quick proposal for macro to make getting a lot of stuff from theme easier. Continue work on design fixes. (Enso Bot - Jun 14, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-14):

**Progress:** Clean the PR for re-review, proposed macro.  Started work on fixing scrollbars. It should be finished by 2022-06-15.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Fix scrollbars and remaining design issues. (Enso Bot - Jun 15, 2022)
<hr />
**Michael Mauderer** reports a new **🔴 DELAY** for yesterday (2022-06-15):

**Summary:** There is **2 days** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 0 days delay** for the delivery of this weekly plan.

**Delay Cause:** Fixing scroll area taking longer than expected. Slow iteration time on fixing design issues. (Enso Bot - Jun 16, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-15):

**Progress:** Clean the PR for re-review, proposed macro.  Started work on fixing scrollbars. It should be finished by 2022-06-17.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Fix scrollbars and remaining design issues. (Enso Bot - Jun 16, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-16):

**Progress:** Fixed scrollbars. Started fixing other layout and colouring issues It should be finished by 2022-06-17.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Finish design fixes and open PR. (Enso Bot - Jun 17, 2022)
<hr />
**Michael Mauderer** reports a new **🔴 DELAY** for the provided date (2022-06-17):

**Summary:** There is **3 days** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 0 days delay** for the delivery of this weekly plan.

2 days weekend.

**Delay Cause:** Additional fixes to scroll area needed (Enso Bot - Jun 20, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for the provided date (2022-06-17):

**Progress:** Worked on the last design fixes for the PR. Last minute realised there is still an issue with the scrollbars interaction left to solve. It should be finished by 2022-06-20.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Fix the scrollbar issue. Open PR. Act on PR feedback. (Enso Bot - Jun 20, 2022)
<hr />
**Michael Mauderer** reports a new **🔴 DELAY** for today (2022-06-21):

**Summary:** There is **2 days** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 0 days delay** for the delivery of this weekly plan.

2 days weekend.

**Delay Cause:** Delay due to more time needed to fix small issues, integrate with ongoing other work and waiting for and applying further review comments. (Enso Bot - Jun 21, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-20):

**Progress:** Fixed scrollbar issues and cleaned up code and design some more. It should be finished by 2022-06-22.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Address final review comments. Resolve merge/integration issues. Do interactive design view on second PR for this task. (Enso Bot - Jun 21, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-21):

**Progress:** Applied final review for 1st PR. Started merge of existing work and found some issues that needed resolving. It should be finished by 2022-06-22.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Merge first PR. Update second PR and mark ready for review. (Enso Bot - Jun 22, 2022)
<hr />
**Michael Mauderer** reports a new **🔴 DELAY** for yesterday (2022-06-23):

**Summary:** There is **2 days** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 0 days delay** for the delivery of this weekly plan.

1 day absence included.

**Delay Cause:** Delay due to CI issue and last minute bug that required fixing. (Enso Bot - Jun 24, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-23):

**Progress:** Encountered some CI issues which needed investigation / resolving before the PR could be merged., Turned out that mergify could noit handle # in the name of a PR. This should be fixed  soon. Also started to look an an issue that appeared in the integration branch about old content hanging around when new content was added. Found the source and a workaround, but will need to investigate the root cause some more as it could indicate a memory leak. It should be finished by 2022-06-24.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Confirm there is no memory leak, or fix it. Then open PR. (Enso Bot - Jun 24, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for yesterday (2022-06-24):

**Progress:** Had some issues building the IDE on a new machine due to an old `cargo-watch` binary. Confirmed there is no memory leak n the current implementation of the ComponentBrowserListViewPanel and expanded the demo scene to show potential issues early on by ensuring that content is set twice. This would reveal issues about removing the first set of items. Following that I cleaned and opened the PR for review. It should be finished by 2022-06-24.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Since I'll be working on the weekend and the review might wait till Monday, I'll try to make the PR smaller by implementing the proposed macro to reduce boilerplate and make future development much faster. (Enso Bot - Jun 25, 2022)
<hr />
**Michael Mauderer** reports a new **🔴 DELAY** for today (2022-06-29):

**Summary:** There is **6 days** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 0 days delay** for the delivery of this weekly plan.

Includes 2 days weekend, 2 days sick,

**Delay Cause:** Too optimistic estimate of PR merge readiness. (Enso Bot - Jun 29, 2022)
<hr />
**Michael Mauderer** reports a new **STANDUP** for today (2022-06-29):

**Progress:** First meeting to talk about design for task #180872321. Was also working on some cleanup of the previous tasks PR, which now adds the new FromTheme macro. Faced some build system issues, which cost quite a bit of time. It should be finished by 2022-06-30.

**Next Day:** Next day I will be working on the [##180892146 task](https://www.pivotaltracker.com/story/show/180892146). Investigate a potential mem leak in the Searcher Panel integration, research and create the task list for #180872321 and, if available, work in PR review feedback. (Enso Bot - Jun 29, 2022)
<hr />
**Acceptance/QA** review set to **revise**

After testing the `component_list_panel_view` demo scene on current `develop` (commit `509663f7a`), I noticed the following discrepancies and confusing observations:

**1. Entries in the rightmost column of Local Scope is not vertically aligned with entries in other columns.**

**NOTE:** Fix in progress at: https://github.com/enso-org/enso/pull/3566

I would expect the right-most column to be either top-aligned, or bottom-aligned; currently, it looks vertically centered.

![](https://user-images.githubusercontent.com/273837/177331141-5fc30704-1d75-45bc-9509-6545395ca9b1.png)

**2. [FIXED](https://github.com/enso-org/enso/pull/3565)** *(was: ~~All groups in the demo scene look gray.~~*)

~~I would expect the demo scene to demonstrate different colors in different component groups.~~

**3. Confusion whether the demo scene shows the "initial" or "updated" Component Browser design.**

Notably:
 - the sections seem ordered like in the ["updated" design](https://discord.com/channels/401396655599124480/986214688163495956);
 - however, section headers are still visible, suggesting the "initial" design;
 - also, no "grayed-out" component groups are visible, suggesting the "initial" design;
 - on the other hand, "grayed-out" component groups are also, in a different way, part of the "initial" design, but this doesn't appear showcased in the demo scene either (see the section: ["Suggesting users most probable first steps"](https://github.com/enso-org/design/blob/main/epics/component-browser/design.md#suggesting-users-most-probable-first-steps) in the "initial" design doc).

**4. All component groups have the same title, making it impossible to verify ordering of sections.**

I would expect the titles to allow verifying the layouting algorithm of groups, being e.g. "Group 1", "Group 2", instead of all groups being titled "Header". Notably, the Task Description suggests that: *"(...) groups should be roughly from left-to-right, from bottom-to-top"* - currently I don't see a way to verify if this was implemented correctly by reviewing the demo scene. (Mateusz Czapliński - Jul 5, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for yesterday (2022-07-05):

**Summary:** There is **5 days** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 1 day delay** for the delivery of this weekly plan.

Weekly Plan Delay is unclear and not certain due to missing standup information for the Sunday 2022-07-03 (see: https://discord.com/channels/401396655599124480/993532885854326965) and MM being currently absent.

**Delay Cause:** Compound delay: seemingly about +1 day by MM (based on available information; see Additional Info); +2 days weekend; +1 day until code review approved and I was pinged to do Acc/QA Review; +1 day until I was available to work on the Acc/QA Review. (Enso Bot - Jul 6, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-07-05):

**Progress:** Did Acc/QA Review, round 1 (result: Revise). It should be finished by 2022-07-05. (Enso Bot - Jul 6, 2022)
<hr />
Ad 3. I would stay with the old design. The dimming of component groups should wait for the optimized version of CB (#182561072) (Adam Obuchowicz - Jul 6, 2022)
<hr />
**Mateusz Czapliński** reports a new **🔴 DELAY** for yesterday (2022-07-06):

**Summary:** There is **1 day** delay in implementation of the [Component List Panel View (#180892146) task](https://www.pivotaltracker.com/story/show/180892146).
It **will cause 1 day delay** for the delivery of this weekly plan.

**Delay Cause:** Submitted fixes for 2 of the issues that I found during the Acc/QA Review of the task. (Enso Bot - Jul 7, 2022)
<hr />
**Mateusz Czapliński** reports a new **STANDUP** for yesterday (2022-07-06):

**Progress:** Implemented and merged PR 3565 fixing a colors issue found during Acc/QA Review. Implemented and opened PR 3566 fixing an issue with vertical alignment of columns in Local Scope (also found during Acc/QA Review). It should be finished by 2022-07-06. (Enso Bot - Jul 7, 2022)
<hr />
**Acceptance/QA** review set to **pass**

With https://github.com/enso-org/enso/pull/3576 and replies in the comments above, all issues found in the previous Acc/QA Review appear addressed and I found no new issues. (Mateusz Czapliński - Jul 12, 2022)
<hr />
